### PR TITLE
#161717320 Provision vof image with ansible

### PR DESCRIPTION
**Pull request by Thegaijin**
#### What does this PR do?
Changes the image provisioning from using bash scripts to primarily using ansible

#### Description of Task to be completed?
- edit docker ci image to include python3 and ansible
- edit base image google-fluentd installation
- edit vof packer file to have ansible as a provisioner
- edit start_vof.sh file to remove setup-scripts functions
- add role for initial vof user setup
- add role to copy files into the image
- add role to setup metricbeat
- add role to setup filebeat
- add role to setup fluentd

#### Any background context you want to provide?
Previously, we were provisioning the images with bash scripts and we came to realise they are harder to debug and for new people to onboard onto the team so we decided to change over to using ansible which is easier to work with.

#### What are the relevant pivotal tracker stories?
[#161717320](https://www.pivotaltracker.com/story/show/161717320)

### CODE REVIEW

File: packer/vof/roles/setup-filebeat/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+- name: create certs directory
+  file:
+    path: /etc/pki/tls/certs
+    state: directory
+
+- name: copy logstash certificate from the gcp bucket
+  command: 'sudo gsutil cp gs://{{ bucket_name }}/elk-ssl/logstash-forwarder.crt /etc/pki/tls/certs/logstash-forwarder.crt'
+
+# - name: copy logstash certificate from the gcp bucket
```
Comment (Thegaijin) [inline packer/vof/roles/setup-filebeat/tasks/main.yml]: Still working on using `gc_storage` have a blocker there.

File: packer/vof/roles/setup-fluentd/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+- name: add staging logs conf file
+  copy:
+    src: templates/vof_staging_logs.conf
+    dest: /etc/google-fluentd/config.d/vof_staging_logs.conf
+
+- name: add vof positioning file
+  copy:
+    src: templates/vof.pos
+    dest: /var/lib/google-fluentd/pos/vof.pos
+# - name: Stop google-fluentd
```
Comment (Thegaijin) [inline packer/vof/roles/setup-fluentd/tasks/main.yml]: Have a blocker here on starting google-fluentd

File: docker/ci/Dockerfile [hunk 3]
```diff
@@ -16,6 +16,10 @@ ENV PATH $PATH:/usr/local/gcloud/google-cloud-sdk/bin
 # install beta components
 RUN gcloud components install beta --quiet
 
+RUN apt-get update && apt-get install python3
+
+RUN apt-get install ansible=1.7.2
```
Comment (orezi) [inline docker/ci/Dockerfile]: Any reason why you have chosen this version? We are looking to have a uniform version of 2.7.0 for ansible across projects @Thegaijin

File: packer/vof/roles/setup-filebeat/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+#   gc_storage:
+#     bucket: '{{ bucket_name }}'
+#     object: elk-ssl/logstash-forwarder.crt
+#     dest: /etc/pki/tls/certs/logstash-forwarder.crt
+#     mode: get
+
+- name: delete filebeat.yml
+  command: rm /etc/filebeat/filebeat.yml
```
Comment (orezi) [inline packer/vof/roles/setup-filebeat/tasks/main.yml]: Make use of file module to remove the file

File: docker/ci/Dockerfile (hunk 3 above)
Comment (Thegaijin) [reply docker/ci/Dockerfile]: I think I missed that, let me change to that version.

File: packer/vof/start_vof.sh [hunk 5]
```diff
@@ -3,11 +3,10 @@
 set -ex
 set -o pipefail
 
-export SCRIPT_FILE="/home/vof/setup-scripts"
-
-# import functions
-. ${SCRIPT_FILE}/setup_filebeat.sh
```
Comment (indungu) [inline packer/vof/start_vof.sh]: To complete this change, please remove the commands [`setup_filebeat`, `setup_metric`] from the `main` function [Lines 341 and 342]

**Pull request merged**